lib/services/search_service.dart

```diff
@@ -0,0 +1,180 @@
+        i++) {
+      File file = files[i];
+      if (file.title.contains(RegExp(query, caseSensitive: false))) {
+        matchedFiles.add(file);
+      }
+    }
+    return matchedFiles;
+  }
+
+  void clearCache() {
+    _cachedFiles.clear();
+  }
+
+  Future<List<LocationSearchResult>> getLocationsAndMatchedFiles(
+    String query,
+  ) async {
+    try {
+      final List<File> allFiles = await SearchService.instance.getAllFiles();
+      final List<LocationSearchResult> locationsAndMatchedFiles = [];
+
+      final response = await _dio.get(
+        _config.getHttpEndpoint() + "/search/location",
+        queryParameters: {"query": query, "limit": 4},
+        options: Options(
+          headers: {"X-Auth-Token": _config.getToken()},
+        ),
+      );
+
+      final matchedLocationNamesAndBboxs =
+          ResultsToListOfPlaceAndBbox.fromMap(response.data);
+
+      for (PlaceAndBbox locationAndBbox
+          in matchedLocationNamesAndBboxs.results) {
+        locationsAndMatchedFiles.add(
+          LocationSearchResult(locationAndBbox.place, []),
+        );
+        for (File file in allFiles) {
+          if (_isValidLocation(file.location)) {
+            //format returned by the api is [lng,lat,lng,lat] where indexes 0 & 1 are southwest and 2 & 3 northeast
+            if (file.location.longitude > locationAndBbox.bbox[0] &&
+                file.location.latitude > locationAndBbox.bbox[1] &&
+                file.location.longitude < locationAndBbox.bbox[2] &&
+                file.location.latitude < locationAndBbox.bbox[3]) {
+              locationsAndMatchedFiles.last.files.add(file);
+            }
+          }
+        }
+      }
+      locationsAndMatchedFiles.removeWhere((e) => e.files.isEmpty);
+      return locationsAndMatchedFiles;
+    } on DioError catch (e) {
+      _logger.info(e);
+      rethrow;
+    }
+  }
+
+  bool _isValidLocation(Location location) {
+    return location != null &&
+        location.latitude != null &&
+        location.latitude != 0 &&
+        location.longitude != null &&
+        location.longitude != 0;
+  }
+
+  // getFilteredCollectionsWithThumbnail removes deleted or archived or
+  // collections which don't have a file from search result
+  Future<List<CollectionWithThumbnail>> getFilteredCollectionsWithThumbnail(
+    String query,
+  ) async {
+    // identify collections which have at least one file as we don't display
+    // empty collection
+
+    final List<File> latestCollectionFiles =
+        await _collectionService.getLatestCollectionFiles();
+    final Map<int, File> collectionIDToLatestFileMap = {
+      for (File file in latestCollectionFiles) file.collectionID: file
+    };
+
+    /* Identify collections whose name matches the search query
+      and is not archived
+      and is not deleted
+      and has at-least one file
+     */
+
+    final List<Collection> matchedCollection = _collectionService
+        .getCollectionIDtoCollections()
+        .values
```

Comment:
If we only care about the values (`List<Collection>`), why call `getCollectionIDtoCollections`? There might be a better API within `CollectionService`?